source/features/useful-not-found-page.tsx

```diff
@@ -28,119 +25,360 @@ async function checkAnchor(anchor: HTMLAnchorElement): Promise<void> {
+		return lastCommitIncludingFile;
+	}
+
+	// Else, check if the file exists in the default branch's history
+	const linkToCommitHistoryOnDefaultBranch =
+		await getLinkToCommitHistoryOnDefaultBranch();
+	if (linkToCommitHistoryOnDefaultBranch) {
+		return {
+			type: "wasOnDefaultBranch",
+			linkToCommitHistory: linkToCommitHistoryOnDefaultBranch,
+		};
+	}
 
+	return undefined;
+}
+
+async function showAdditionalInformation(bar: Element): Promise<void> {
+	// Check what happened
+	const eventThatHappened = await whatHappenedToTheFile();
+	if (!eventThatHappened) {
+		return;
+	}
+
+	// Parse that and do/show something
+	if (eventThatHappened.type === "isOnDefaultBranch") {
+		bar.after(
+			<p className="container mt-4 text-center">
+				<a href={eventThatHappened.linkToLastVersion}>This file</a> only exists
+				on the default branch.
+			</p>
+		);
+		return;
+	}
+
+	if (eventThatHappened.type === "wasOnDefaultBranch") {
+		bar.after(
+			<p className="container mt-4 text-center">
+				This file used to exist on the default branch - see{" "}
+				<a href={eventThatHappened.linkToCommitHistory}>commit history</a>.
+			</p>
+		);
+		return;
+	}
+
+	if (eventThatHappened.type === "removed") {
+		bar.after(
+			<p className="container mt-4 text-center">
+				<a href={eventThatHappened.linkToLastVersion}>This file</a> was deleted
+				by{" "}
+				<a href={eventThatHappened.linkToCommit}>
+					{eventThatHappened.commitSha}
+				</a>{" "}
+				(<relative-time datetime={new Date(eventThatHappened.commitDate!)} />) -
+				see <a href={eventThatHappened.linkToCommitHistory}>commit history</a>.
+			</p>
+		);
+		return;
+	}
+
+	if (eventThatHappened.type === "renamed") {
+		bar.after(
+			<p className="container mt-4 text-center">
+				<a href={eventThatHappened.linkToLastVersion}>This file</a> was renamed
+				to{" "}
+				<a href={eventThatHappened.linkToNewVersion}>
+					{eventThatHappened.newFilePath}
+				</a>{" "}
+				by{" "}
+				<a href={eventThatHappened.linkToCommit}>
+					{eventThatHappened.commitSha}
+				</a>{" "}
+				(<relative-time datetime={new Date(eventThatHappened.commitDate!)} />) -
+				see <a href={eventThatHappened.linkToCommitHistory}>commit history</a>.
+			</p>
+		);
+	}
+}
+
+async function showWhatIsMissingInPath(bar: Element): Promise<void> {
+	// Get the current URL
+	const parts = parseCurrentURL();
+
+	// Display the path of the file (including repo owner, repo name and branch name)
 	for (const [i, part] of parts.entries()) {
-		if (i === 2 && ['tree', 'blob', 'edit'].includes(part)) {
-			// Exclude parts that don't exist as standalones
+		// Exclude parts that don't exist as standalones
+		if (i === 2 && ["tree", "blob", "edit"].includes(part)) {
 			continue;
 		}
 
+		// The last part of the URL is a known 404
 		if (i === parts.length - 1) {
-			// The last part of the URL is a known 404
-			bar.append(' / ', getStrikeThrough(part));
+			// Put a strikthrough through it by default
```

Comment:
Other unrelated change. Since this is a huge PR, refer from changing code unnecessarily

Reply:
Sure, I'll revert those comments.